### PR TITLE
Stage yeet commands fixing

### PR DESCRIPTION
**Pull request by hymm**
# Objective

Fix a few bugs around command application

1. final command application is running on executor thread instead of schedule thread
2. unapplied systems buffer was never being cleared

## Solution

1. move the final command application outside of executor and outside of the scope
2. clear the unapplied_systems buffer on self instead of the cloned one.
3.  snuck removes some more spans into this pr. These spans aren't needed as they duplicate ones on Commands::apply

### CODE REVIEW

File: crates/bevy_ecs/src/scheduling/executor/multi_threaded.rs [hunk 1]
```diff
@@ -212,6 +200,20 @@ impl SystemExecutor for MultiThreadedExecutor {
                 scope.spawn(executor);
             },
         );
+
+        // SAFETY: all systems have completed
+        let world = unsafe { &mut *world.get() };
+        // this is running on the executor thread when it should run on the scope's thread
+        apply_system_buffers(&mut self.unapplied_systems, systems, world);
+        self.unapplied_systems.clear();
```
Comment (maniwani) [inline crates/bevy_ecs/src/scheduling/executor/multi_threaded.rs]: Since you've fixed it, probably reword this as an invariant.

Comment (alice-i-cecile) [reply crates/bevy_ecs/src/scheduling/executor/multi_threaded.rs]: Agreed on this comment.

Comment (alice-i-cecile) [reply crates/bevy_ecs/src/scheduling/executor/multi_threaded.rs]:
```suggestion
        // SAFETY: all systems have completed, and so no outstanding accesses remain
        let world = unsafe { &mut *world.get() };
        // Commands should be applied while on the scope's thread, not the executor's thread
        apply_system_buffers(&mut self.unapplied_systems, systems, world);
        self.unapplied_systems.clear();
```
